inference/predictor_deployment.py

```diff
@@ -145,33 +146,71 @@ async def __call__(self, http_request: Request) -> Union[StreamingResponse, JSON
                 self.consume_streamer_async(streamer), status_code=200, media_type="text/plain"
             )
 
-    async def stream_response(self, prompt, config):
+    async def openai_call(self, prompt, config, streaming_response=True):
         prompts = []
         if isinstance(prompt, list):
-            if self.process_tool is not None:
-                prompt = self.process_tool.get_prompt(prompt)
-                prompts.append(prompt)
+            is_chat, is_prompts = get_input_format(prompt)
+            if is_chat:
+                if self.process_tool is not None:
+                    prompt = self.process_tool.get_prompt(prompt)
+                    prompts.append(prompt)
+                else:
+                    prompts.extend(prompt)
+            elif is_prompts:
+                yield HTTPException(
+                    400, "mulitple prompts is not supported now when using openai compatible api."
+                )
             else:
-                prompts.extend(prompt)
+                yield HTTPException(400, "invalid prompt format.")
         else:
             prompts.append(prompt)
 
-        if self.use_deepspeed:
-            self.predictor.streaming_generate(prompts, self.streamer, **config)
-            response_handle = self.consume_streamer()
-        else:
-            streamer = self.predictor.get_streamer()
-            self.loop.run_in_executor(
-                None,
-                functools.partial(self.predictor.streaming_generate, prompts, streamer, **config),
-            )
-            response_handle = self.consume_streamer_async(streamer)
-        async for output in response_handle:
+        if not streaming_response:
+            if self.use_vllm:
+                generate_result = (await self.predictor.generate_async(prompts, **config))[0]
+                generate_length = generate_result.generate_length
+            else:
+                generate_result = self.predictor.generate(prompts, **config)
+            input_length = generate_result.input_length
```

Comment:
I think you can use generate_result.input_length directly, no need to introduce new variable